File: 백준/Bronze/10813. 공 바꾸기/공 바꾸기.js
const fs = require("fs");
const path = require("path");
const input = fs
  .readFileSync(
    process.platform === "linux"
      ? "/dev/stdin"
      : path.join(__dirname, "/input.txt"),
  )
  .toString()
  .trim()
  .split("\n");

const solution = (input) => {
  const [N, M] = input[0].split(" ").map(Number);

  const arr = Array.from({ length: N }, (_, k) => k + 1);
  for (let i = 1; i <= M; i++) {
    const [n1, n2] = input[i].split(" ").map((v) => Number(v) - 1);
    const temp = arr[n1];
    arr[n1] = arr[n2];
    arr[n2] = temp;
  }
  console.log(arr.join(" "));
};

solution(input);

File: 백준/Bronze/10813. 공 바꾸기/공 바꾸기.cc
#include <iostream>

using namespace std;

int main()
{
    int N,M;

    cin >> N >> M;
    
    int arr[N+1];
    for(int i=1;i<=N;i++){
        arr[i]=i;
    }
    for(int i=0;i<M;i++){
        int a,b;
        cin >> a >> b;
        int temp;
        temp=arr[a];
        arr[a]=arr[b];
        arr[b]=temp;
    }
    
    for(int i=1;i<=N;i++){
        cout << arr[i] << " ";
    }
    return 0;
}